master/internal/db/postgres_groups.go

```diff
@@ -57,17 +68,18 @@ func (db *PgDB) AddUsersToGroup(ctx context.Context, gid int, uids ...model.User
 		})
 	}
 
-	_, err := Bun().NewInsert().Model(&groupMem).Exec(ctx)
-	return err
+	res, err := Bun().NewInsert().Model(&groupMem).Exec(ctx)
+
+	return checkIfFound(res, err)
```

Comment:
Does this return `ErrorNotFound` when the group and/or any of the users is not found? That makes me think of an edge case we should probably test: if this operation fails, we want the system to behave in a predictable way, either rolling back the whole transaction or keeping around the few that did succeed. Either way, we should have a test for this behavior once we decide how it should work. In my opinion, the whole transaction should succeed or fail. What do you think?

Reply:
At a cursory look, it seems like bun provides a transaction abstraction that we might be able to take advantage of. I think if we wrap some things in a transaction we can automatically rollback in the event of an error?